### PR TITLE
html / js / python results generation

### PR DESCRIPTION
**Pull request by drewkerrigan**
Due to difficulties getting R intstalled (specifcally gfortran) on certain platforms, it has been requested that we create a non-R based report option for basho_bench results. 

### Usage

1. After running a test, use the following make target:

    ```
    make results-browser
    ```

    Doing so will start a minimal python SimpleHTTPServer based script: `Serving at: http://localhost:8000`

2. Navigating to [http://localhost:8000](http://localhost:8000) will display a series of [Google Charts](https://developers.google.com/chart/) based graphs that mimic the ones displayed by the older R generated graphs.

    Example Graphs

3. Visiting [http://localhost:8000](http://localhost:8000) for the first time will perform an ajax request to the python script  located at `priv/results-browser.py` which saves all of the graphs drawn to a hidden canvas element and converted to PNG. The resulting image file is located at tests/current/summary.png. 

    The summary.png is obviously a static file, but the HTML generated by [http://localhost:8000](http://localhost:8000) is a set of live charts which allows for highlighting specific datapoints.

### Dependencies

Currently using this method of report generation requires a basic ***python*** installation (no package deps outside of the base install), as well as ***access to the internet*** to allow dynamic JS loading from `https://www.google.com/jsapi`. 

The access to the internet requirement could probably be eliminated by copying the required modules from google in the future. 

The python requirement could probably be eliminated too with a CGI based web server, but python seems to be a common enough utility on most systems in my experience.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,4 +7,4 @@ package
 .rebar
 *~
 #*#
-
+.DS_Store
```
Comment (sargun) [inline .gitignore]: Ideally, this should be a separate commit.

File: priv/results-browser.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import SimpleHTTPServer
```
Comment (sargun) [inline priv/results-browser.py]: Prefix with #!env python

File: priv/results-browser.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import SocketServer
+import logging
+import cgi
+import base64
+
```
Comment (sargun) [inline priv/results-browser.py]: import argparse

File: priv/results-browser.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import sys
```
Comment (sargun) [inline priv/results-browser.py]: --import sys

File: priv/results-browser.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+if len(sys.argv) > 2:
+    PORT = int(sys.argv[2])
+    I = sys.argv[1]
+elif len(sys.argv) > 1:
+    PORT = int(sys.argv[1])
+    I = ""
+else:
+    PORT = 8000
+    I = ""
+
```
Comment (sargun) [inline priv/results-browser.py]: The earlier should be handled by argparse

File: priv/results-browser.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    def do_POST(self):
+      logging.warning("======= POST STARTED =======")
+      length = self.headers['content-length']
+      data = self.rfile.read(int(length))
+
+      with open("./" + self.path, 'w') as fh:
+        fh.write(base64.b64decode(data.decode()))
+
+      self.send_response(200)
+
```
Comment (sargun) [inline priv/results-browser.py]: Use name == '__main__':

```

def startServer(host, port):
  handler = ServerHandler
  httpd = SocketServer.TCPServer(("", PORT), Handler)
  print 'Serving at: http://{host}:{port}'.format(host=host, port=port)
  httpd.serve_forever()
  
if __name__ == '__main__':
  parser = argparse.ArgumentParser(description='Results generator')
  parser.add_argument('--port', '-p', type=int, help='Port for results generator to bind to', default=8080, required=False)
  parser.add_argument('--host', type=str, help='Host for results generator to bind to', default='localhost', required=False)
  args = parser.parse_args()
  startServer(args.host, args.port)
```

File: priv/results-browser.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+      logging.warning("======= GET STARTED =======")
+      logging.warning(self.headers)
+      SimpleHTTPServer.SimpleHTTPRequestHandler.do_GET(self)
+
+    def do_POST(self):
+      logging.warning("======= POST STARTED =======")
+      length = self.headers['content-length']
+      data = self.rfile.read(int(length))
+
+      with open("./" + self.path, 'w') as fh:
```
Comment (sargun) [inline priv/results-browser.py]: You really should use os.path.join here, but, alas.

File: priv/results-browser.py (hunk 2 above)
Comment (drewkerrigan) [inline priv/results-browser.py]: Done

File: Makefile [hunk 8]
```diff
@@ -65,6 +65,9 @@ mb_sec-results:
 mib_sec-results:
 	Rscript --vanilla priv/summary.r --ylabel1stgraph MiB/sec -i tests/current
 
+results-browser:
+	cp -R priv/results-browser/* tests/current && cd tests/current && python -c 'import os, json; print json.dumps(os.listdir("."))' > web/data.json && python ../../priv/results-browser.py --host localhost --port 8080
```
Comment (sargun) [inline Makefile]: Nitpicking, but this doesn't need --host and --port, since those are defaults
